### PR TITLE
scr/pytohn/CRABInterface supports py3

### LINKED ISSUE #6830: src/python/CRABInterface should support py3
src/python/CRABInterface should run in py3

Status quo:

- [x] crabserver REST interface run from dmwm/python3 branch
- [x] CMSKubernetes: use https://github.com/dmwm/CMSKubernetes/pull/867
- [x] cmsdist: use https://github.com/cms-sw/cmsdist/pull/7452

known problems and missing changes
- [x] https://cmsweb-test1.cern.ch/crabserver/ui returns 404. missing index.html file. Note from Stefano: this is known to happen whenever running cranserver from GH repos instead of `current`. Not worth looking into a fix.
- [x] (1) `tm_edm_outfiles': [b'minbias.root']`. we have a workaround https://github.com/dmwm/CRABServer/releases/tag/py3.211104 (1.0), or another workaround in https://github.com/mapellidario/CRABServer/tree/6830-fix-1  (1.1)
- [x]  (2) TypeError: cannot use a string pattern on a bytes-like object (https://github.com/dmwm/CRABServer/issues/6830#issuecomment-961416487) -> testing a fix in https://github.com/mapellidario/CRABServer/tree/6830-fix-1
- [x] (3) addStatusAndFailure(result, status = 'UNKNOWN', failure = ee.info) (https://github.com/dmwm/CRABServer/issues/6830#issuecomment-961416487) -> testing a fix in https://github.com/mapellidario/CRABServer/tree/6830-fix-1
- [x] (4) `response.headers['ETag'] = '"%s"' % hashlib.sha1(result).hexdigest()` (appears only when opeing the web ui):
  - fixed upstream: https://github.com/dmwm/WMCore/pull/10917
- [x] (5) `b"The following sites from the user site whitelist are blacklisted by the CRAB server` -> testing a [fix](https://github.com/mapellidario/CRABServer/blob/51d1837703e8a11c480f02d0e86b476e0840b65f/src/python/CRABInterface/HTCondorDataWorkflow.py#L228) in https://github.com/mapellidario/CRABServer/tree/6830-fix-1
- [x]  (6) decoding problem when wrong run list (https://github.com/dmwm/CRABServer/issues/6830#issuecomment-975730554)
- [x] py3 crabserver compatible with py2-submitted tasks
  - [x] (7) https://github.com/dmwm/CRABServer/compare/python3...mapellidario:6830-fix-4

| taskname | submission env | status | changes |
|-|-|-|-|
| `211207_160318:dmapelli_crab_20211207_170313` | test2 (py2) | submitted | up to 7 |
| `211207_162346:dmapelli_crab_20211207_172342` | test2 (py2) | submitted | up to 7 |
| `211207_163112:dmapelli_crab_20211207_173108` | test2 (py2) | submitted | up to 7 |
| https://cmssdt.cern.ch/dmwm-jenkins/job/CRABServer_ExecuteTests/490/ | test1 (py3) | ?? | up to 7 |
| https://cmssdt.cern.ch/dmwm-jenkins/job/CRABServer_ExecuteTests/491/ | test2 (py2) | ?? | up to 7 |

tests: (old tests removed from here)

### PR DESCRIPTION
**Pull request by mapellidario**
Fixed #6830 

#### overall information

Status: Work in progress

This PR should be merged after merging #6828 

This PR does **not** support py2.

#### details

TDL:
- [x] futurize (all changes suggested by futurize have been vetted. many have been discarded as not necessary, some have been implemented)
- [x] `pylint --py3k` (pylint does not report problems regarding py3 compatibility)
- [x] manual check of known problems that futurize and pylint do not see (`pickle`, `open()` files from disk, access to exception properties, ...)

Open questions:

- [ ] RESTTask.y: use of `str()`
- [ ] Utilities.py: use of `io.BytesIO()` . do we need to convert back yo py3 `str`?

### CODE REVIEW

File: src/python/CRABInterface/RESTWorkerWorkflow.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 from CRABInterface.Utilities import getDBinstance
 from CRABInterface.RESTExtensions import authz_login_valid
 from CRABInterface.Regexps import (RX_TASKNAME, RX_WORKER_NAME, RX_STATUS, RX_TEXT_FAIL, RX_SUBPOSTWORKER,
-                                  RX_SUBGETWORKER, RX_JOBID)
+                                  RX_JOBID)
```
Comment (belforte) [inline src/python/CRABInterface/RESTWorkerWorkflow.py:13]: why remove SUBGETWORKER but not SUBPOSTWORKER and what has this to do with panda ?
Can you elaborate a  bit on the rational ?

Comment (mapellidario) [reply src/python/CRABInterface/RESTWorkerWorkflow.py:13]: Your commit https://github.com/belforte/CRABServer/commit/5a8fa719c707f58a5cfa688e3eb288c042bedc5d#diff-f457a331093710562f796332dd9f5ba55f470f9c1adf1adb4048b1c7e32e20f9
removed `RX_SUBGETWORKER` from `CRABInterface.Regexps`, i could not do anything but removing the import and not using it

Comment (belforte) [reply src/python/CRABInterface/RESTWorkerWorkflow.py:13]: thanks Dario. Sorry I did not look. I think I understand now. Well... things could have been named better to begin with. Let me review things in detail. We do not use the `jobgroup` table in DB anymore.

Comment (belforte) [reply src/python/CRABInterface/RESTWorkerWorkflow.py:13]: OK. indeed `subresource` is not used for get and you should kindly remove `subresource` from L90 to make it clear.

Comment (belforte) [reply src/python/CRABInterface/RESTWorkerWorkflow.py:13]: this is a cleanup which could have been done long ago :-(

**Review by belforte — APPROVED** [review]

**Comment by mapellidario** [comment]
In the meantime, we have some progress!

```
> crab submit -c Pythia.py

Will use CRAB configuration file Pythia.py
Importing CMSSW configuration pset_MC.py
Finished importing CMSSW configuration pset_MC.py
Sending the request to the server at cmsweb-test1.cern.ch
Success: Your task has been delivered to the dev CRAB3 server.
Task name: 211102_141303:dmapelli_crab_20211102_151153
Project dir: crab_projects/crab_20211102_151153
Please use ' crab status -d crab_projects/crab_20211102_151153 ' to check how the submission process proceeds.
Log file is /afs/cern.ch/user/d/dmapelli/crab/issue/crabserver-restpy3/CMSSW_10_6_18/src/crab_projects/crab_20211102_151153/crab.log

> crab status -d crab_projects/crab_20211102_151153 
CRAB project directory:         /afs/cern.ch/user/d/dmapelli/crab/issue/crabserver-restpy3/CMSSW_10_6_18/src/crab_projects/crab_20211102_151153
Task name:                      211102_141303:dmapelli_crab_20211102_151153
Grid scheduler - Task Worker:   crab3@vocms059.cern.ch - crab-dev-tw01
Status on the CRAB server:      HOLDING on command SUBMIT
Task URL to use for HELP:       https://cmsweb-test1.cern.ch/crabserver/ui/task/211102_141303%3Admapelli_crab_20211102_151153
Dashboard monitoring URL:       https://monit-grafana.cern.ch/d/cmsTMDetail/cms-task-monitoring-task-view?orgId=11&var-user=dmapelli&var-task=211102_141303%3Admapelli_crab_20211102_151153&from=1635858783000&to=now
Log file is /afs/cern.ch/user/d/dmapelli/crab/issue/crabserver-restpy3/CMSSW_10_6_18/src/crab_projects/crab_20211102_151153/crab.log

```

using the REST with python3 branch! :tada:

**Pull request merged**